File: YuGiOh2/Cards/C65396880.cs
using YuGiOh2.Base;

namespace YuGiOh2.Cards
{
    /// <summary>
    /// 大革命
    /// </summary>
    public class C65396880
    {
        public static int Type { get; } = (int)AffectMomentType.WhenAttacked;

        public static void ProcessEffect(string targetID, Player player)
        {
            for (int i = 0; i < 5; i++)
            {
                if (player.Field.MonsterFields[i] == null || 
                    player.Field.MonsterFields[i].Status.DefensePosition ||
                    player.Field.MonsterFields[i].Level > 3)
                    continue;

                Destroy(player.Enemy);
                return;
            }
            Destroy(player);
        }

        private static void Destroy(Player player)
        {
            for (int i = 0; i < 5; i++)
            {
                if (player.Field.MonsterFields[i] != null)
                {
                    player.AddCardToGrave(player.Field.MonsterFields[i]);
                    player.Field.MonsterFields[i] = null;
                }
                if (player.Field.SpellAndTrapFields[i] != null)
                {
                    player.AddCardToGrave(player.Field.SpellAndTrapFields[i]);
                    player.Field.SpellAndTrapFields[i] = null;
                }
            }
            if (player.Field.FieldField != null)
            {
                if (!player.Field.FieldField.Status.FaceDown)
                {
                    player.Enemy.ProcessEnemyField();
                }
                else
                {
                    player.Grave.Add(player.Field.FieldField);
                    player.Field.FieldField = null;
                }
            }
            player.DiscardHands(player.Hands.Count);
            player.TrapsWhenAttack.Clear();
            player.TrapsWhenSummon.Clear();
        }
    }
}

File: YuGiOh2/Scripts/C65396880.lua
Type = 2;
--大革命
function ProcessEffect(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil and not monster.Status.DefensePosition and monster.Level < 4 then
            destroy(player.Enemy);
            return ;
        end
    end
    destroy(player);
end

function destroy(player)
    for i, monster in pairs(player.Field.MonsterFields) do
        if monster ~= nil then
            player:AddCardToGrave(monster);
            player:ClearMonsterField(i - 1);
        end
    end
    for i, spell in pairs(player.Field.SpellAndTrapFields) do
        if spell ~= nil then
            player:AddCardToGrave(spell);
            player:ClearSpellAndTrapField(i - 1);
        end
    end
    player:DiscardHands(player.Hands.Count);
    player.TrapsWhenAttack:Clear();
    player.TrapsWhenSummon:Clear();
end